internal/core/command/put.go

```diff
@@ -44,5 +50,7 @@ func NewPutCommand(
 		request.Header.Set(clients.CorrelationHeader, correlationID.(string))
 	}
 
+	request.Header.Set(clients.ContentType, r.Header.Get(clients.ContentType))
```

Comment:
Should we be passing more headers than just `Content-type` and `Correlation`? I am thinking we pass on all the headers from the original request since we are not sure which ones might be useful. For example, `Accept`, `Content-MD5`, `Date`, `Host`, etc. We discussed this previously, did you do any digging and find if there are any headers you thought were important to propagate? I think we maybe able to introduce the [Forwarded](https://tools.ietf.org/html/rfc7239#section-4) header in our request to properly inform the requester as of the origin of the request, just a thought.

Reply:
- Yes, the point regarding whether we should be passing more headers than just `Content-type` and `Correlation` is very pertinent, especially given the open-ended nature of which headers may be useful to end users. I did do some digging,  most recently in connection with the possibility of introducing the [Forwarded header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded) in our request.
- So far, it seems like the golang community is grappling with [an issue in this area](https://github.com/golang/go/issues/20526), _without_ a clear outcome or direction.

Reply:
Per Core WG 26-Mar-2020 we will not add these additional headers for pass-through. A separate task would be to audit passing of Content-Length on all requests. This may be handled already by the Go http implementation.

Reply:
- To add a data point regarding the mention above of the `Content-Length` part of the header (i.e. `This may be handled already by the Go http implementation.`), on looking up the `golang` reference online for [Package http](https://golang.org/pkg/net/http/):
```
    // For client requests, certain headers such as Content-Length
    // and Connection are automatically written when needed and
    // values in Header may be ignored. See the documentation
    // for the Request.Write method.
```

Reply:
- As @AnthonyMBonafide pointed out during the aforementioned `Core WG` 26-Mar-2020 call, we're specifically talking here in the context of the `core-command` talking to a device service. Noting, too, that device service won't do anything with those headers; they are not "materialized".